sql/4_1_2-to-4_2_0_upgrade.sql

```diff
@@ -131,6 +131,65 @@ CREATE TABLE `erx_ttl_touch` (
 ENGINE = InnoDB COMMENT = 'Store records last update per patient data process';
 #EndIf
 
+#IfNotTable care_plan
+CREATE TABLE IF NOT EXISTS `care_plan` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `date` date NOT NULL,
+  `encounter` int(10) NOT NULL,
+  `pid` int(11) NOT NULL,
+  `diag_1` varchar(35) NOT NULL,
+  `diag_2` varchar(35) NOT NULL,
+  `diag_3` varchar(25) NOT NULL,
+  `diag_4` varchar(25) NOT NULL,
+  `diag_5` varchar(25) NOT NULL,
+  `active_1` varchar(2) NOT NULL,
+  `active_2` varchar(2) NOT NULL,
+  `active_3` varchar(2) NOT NULL,
+  `active_4` varchar(2) NOT NULL,
+  `active_5` varchar(2) NOT NULL,
+  `risk_1` varchar(255) NOT NULL,
+  `risk_2` varchar(255) NOT NULL,
+  `risk_3` varchar(35) NOT NULL,
+  `risk_4` varchar(255) NOT NULL,
+  `risk_5` varchar(255) NOT NULL,
+  `assessment_1` varchar(255) NOT NULL,
+  `assessment_2` varchar(255) NOT NULL,
+  `assessment_3` varchar(35) NOT NULL,
+  `assessment_4` varchar(255) NOT NULL,
+  `assessment_5` varchar(255) NOT NULL,
+  `goal_1` varchar(255) NOT NULL,
+  `goal_2` varchar(255) NOT NULL,
+  `goal_3` text NOT NULL,
+  `goal_4` varchar(255) NOT NULL,
+  `goal_5` varchar(255) NOT NULL,
+  `provider_1` varchar(30) NOT NULL,
+  `provider_2` varchar(30) NOT NULL,
+  `provider_3` int(3) NOT NULL,
+  `provider_4` int(3) NOT NULL,
+  `provider_5` int(3) NOT NULL,
+  `prevention` text NOT NULL,
+  `pmh` text NOT NULL,
+  `psh` varchar(255) NOT NULL,
+  `fhsh` text NOT NULL,
+  `sh` varchar(255) NOT NULL,
+  `audit` int(1) NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id` (`id`),
+  KEY `pid` (`pid`)
+) ENGINE=InnoDB  DEFAULT CHARSET=utf8 AUTO_INCREMENT=66 ;
```

Comment:
Rather than hard-code a maximum of 5 for each of the items, wouldn't a database structure that allows as many as want be better. For example, a care_plan_element table row would contain a diag, active, risk, etc, and a mapping id back to the id of the care_plan table element, which houses the main things like id, date, encounter.